CONTRIBUTING.md

```diff
@@ -65,3 +64,30 @@ should be triaged immediately upon open by the PR author.
     community members. Be reasonable.
 - All PRs should be labelled with the current milestone before merging. If a PR for an issue
     labelled with a different milestone is to be merged, update the issue milestone as well.
+
+## Developing Wrangler
+
+### Get started
+
+To get started with developing Wrangler, we recommend that you first get [up to speed with Wrangler](https://developers.cloudflare.com/workers/quickstart). Then, get up to speed with the [basics of Rust](https://www.rust-lang.org/learn/get-started). (You'll need to install `rustup` where we're going).
+
+### Build Wrangler from source
+
+To build Wrangler from source, fork the repo, clone it, and `cd wrangler`.
+
+There are many ways to build and execute a development version of Wrangler:
+
+`cargo build` will produce a local binary at `./target/debug/wrangler` that you can execute.
+`cargo install --debug --path .` will replace any globally installed wrangler with the one you've just built from source.
+
+You can read more about cargo [here](https://doc.rust-lang.org/cargo/), just find something that works for you.
+
+### Command-line argument parsing
+
+The primary framework we use for developing Wrangler is called [clap](https://clap.rs), which provides fast and structured argument parsing. This is how different features are exposed to users, and most of that logic lives in [main.rs](./src/main.rs).
+
+Each of the commands supported by Wrangler have entrypoints in [./src/commands](./src/commands). It's useful to understand the [module system](https://doc.rust-lang.org/rust-by-example/mod.html) if you will be adding new commands or need to refactor/organize imports.
+
```

Comment:
it is also nice to call out pre-push steps, like test, fmt, and clippy, and the fact that if CI fails your PR may not be reviewed (if it is a legit bug). i also wouldn't mind calling out that additions to the readme/testing suite should be considered part of Definition of Done, but maybe we need to hash that out and add it later.